saleor/graphql/core/dataloaders.py

```diff
@@ -50,3 +50,17 @@ def batch_load_fn(  # pylint: disable=method-hidden
 
     def batch_load(self, keys: Iterable[K]) -> Union[Promise[List[R]], List[R]]:
         raise NotImplementedError()
+
+
+class DataLoaderWithSort(DataLoader):
```

Comment:
I don't think that we need this. 
You can have simple `class MediaByProductIdLoaderSort(DataLoader)` and then in `resolve_media`
```
    @staticmethod
     def resolve_media(root: ChannelContext[models.Product], info, sort_by=None):
         if sort_by is None:
             sort_by = {
                 "field": MediaChoicesSortField.ID.value,  # type: ignore
                 "direction": "",
             }
         def sort_media(media):
             ...
             return sorted_media
         return MediaByProductIdLoader(info.context, sort_by).load(root.node.id).then(sort_media)
```

With current approach, If someone use normal `Media` dataloader and `MediaWithSort` in the same query, then the loader will fetch the data twice. This is why we should use the same dataloader.